internal/backends/clojure/deps.go

```diff
@@ -0,0 +1,504 @@
+		return MavenVersionInfoOp{}, err
+	}
+	metadata := MavenVersionInfoOp{}
+	xml.Unmarshal(bs, &metadata)
+	return metadata, nil
+}
+
+func GetMPackageInfoMaven(pkg JavaPackage) (MPkgInfo, error) {
+	metadata, err := MavenGetVersionInfo(pkg)
+	if err != nil {
+		return MPkgInfo{}, err
+	}
+	versions := []string{}
+	for _, entry := range metadata.Versioning.Versions.VersionList {
+		versions = append(versions, entry)
+	}
+	path := strings.Replace(pkg.toString(), ".", "/", -1)
+	url := "http://search.maven.org/maven2/" + path + "/" + metadata.Versioning.LatestVersion + "/" + pkg.Artifact + "-" + metadata.Versioning.LatestVersion + ".pom"
+	resp, err := http.Get(url)
+	if err != nil || resp.StatusCode != 200 {
+		return MPkgInfo{}, errors.New("Couldn't fetch package info")
+	}
+	bs, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return MPkgInfo{}, err
+	}
+
+	pkgPom := MavenFetchOp{}
+	xml.Unmarshal(bs, &pkgPom)
+	var author string
+	if len(pkgPom.Developers.Developer) > 0 {
+		author = pkgPom.Developers.Developer[0].Name
+	}
+	return MPkgInfo{
+		Id:            pkg,
+		LatestVersion: metadata.Versioning.LatestVersion,
+		Versions:      versions,
+		Description:   pkgPom.Description,
+		ScmUrl:        pkgPom.Scm.Url,
+		Author:        author,
+	}, nil
+}
+
+func GetMPackageInfo(pkg JavaPackage) (MPkgInfo, error) {
+	minfo, err := GetMPackageInfoMaven(pkg)
+	if err == nil {
+		return minfo, nil
+	} else {
+		cinfo, err := GetMPackageInfoClojars(pkg)
+		if err == nil {
+			return cinfo, nil
+		}
+		return MPkgInfo{}, errors.New("Couldn't fetch info for package: " + pkg.toString() + " from any sources")
+	}
+}
+
+func GenPkgInfo(pinfo MPkgInfo) api.PkgInfo {
+	return api.PkgInfo{
+		Name:             pinfo.Id.Group + "/" + pinfo.Id.Artifact,
+		Description:      pinfo.Description,
+		Version:          pinfo.LatestVersion,
+		HomepageURL:      pinfo.Homepage,
+		DocumentationURL: "",
+		SourceCodeURL:    "",
+		BugTrackerURL:    "",
+		Author:           pinfo.Author,
+		License:          "",
+		Dependencies:     nil,
+	}
+}
+
+func GetPackageInfo(pkg JavaPackage) (api.PkgInfo, error) {
+	pinfo, err := GetMPackageInfo(pkg)
+	if err != nil {
+		return api.PkgInfo{}, err
+	}
+	return GenPkgInfo(pinfo), nil
+}
+
+func findPackagesClojars(name string) ([]api.PkgInfo, error) {
+	s := "https://clojars.org/search?q=" + name + "&format=json"
+	res, err := http.Get(s)
+	if err != nil || res.StatusCode != 200 {
+		return nil, errors.New("Couldn't fetch package info")
+	}
+	var resp ClojarsSearchOp
+	bs, err := ioutil.ReadAll(res.Body)
+	err = json.Unmarshal(bs, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	fres := []api.PkgInfo{}
+	for _, p := range resp.Results {
+		pinfo, err := GetMPackageInfoClojars(JavaPackage{
+			Artifact: p.JarName,
+			Group:    p.GroupName,
+		})
+		if err != nil {
+			panic(err)
+		}
+		fres = append(fres, GenPkgInfo(pinfo))
+	}
+
+	return fres, nil
+}
+
+func findPackagesMaven(name string) ([]api.PkgInfo, error) {
```

Comment:
Neat! I adapted some code to search Maven for the `java-maven` backend: https://github.com/replit/upm/blob/master/internal/backends/java/msch.go - I like what you've written here though and in a follow-on change I may point the `java-maven` backend at these structs & functions!